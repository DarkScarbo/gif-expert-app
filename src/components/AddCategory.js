import React, { useState } from 'react';
import Proptypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => setInputValue(e.target.value);
  const handleSubmit = (e) => {
    e.preventDefault();
    setCategories(ctgs => [inputValue, ...ctgs]);
    setInputValue("")
  };

    return (
        <form onSubmit={handleSubmit}>
          <input
              type="text"
              value={inputValue}
              onChange={handleInputChange}
              required
          /> 
        </form>
    )
}

AddCategory.propTypes = {
  setCategories: Proptypes.func.isRequired
 }