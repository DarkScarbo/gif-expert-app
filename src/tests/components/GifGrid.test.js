import React from 'react'
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

const { shallow } = require("enzyme")

describe('Tests in <GifGrid />', () => {

   const category = 'Rupaul'
   const gifs = [{
       id: 123,
       url: 'https://localhost',
       title: 'Rupaul'
    },{
        id: 345,
        url: 'https://localhost',
        title: 'Drag Queen'
    }];

    test('should show the component correctly', () => {  
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={ category } />)
    
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show items after loading them with useFetchGifs', () => {
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
