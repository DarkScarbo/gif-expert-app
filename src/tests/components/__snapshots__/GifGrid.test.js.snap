// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> should show items after loading them with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rupaul
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={123}
      key="123"
      title="Rupaul"
      url="https://localhost"
    />
    <GifGridItem
      id={345}
      key="345"
      title="Drag Queen"
      url="https://localhost"
    />
  </div>
</Fragment>
`;

exports[`Tests in <GifGrid /> should show the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rupaul
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
